api-admin/src/main/java/bio/terra/pearl/api/admin/service/enrollee/EnrolleeSearchExtService.java

```diff
@@ -52,4 +54,21 @@ public Map<String, SearchValue.SearchValueType> getExpressionSearchFacets(
     return this.enrolleeSearchService.getExpressionSearchFacetsForStudyEnv(
         studyEnvironment.getId());
   }
+
+  public List<EnrolleeSearchExpressionResult> executeSearchExpression(
+      AdminUser operator,
+      String portalShortcode,
+      String studyShortcode,
+      EnvironmentName envName,
+      String expression) {
+
+    authUtilService.authUserToStudy(operator, portalShortcode, studyShortcode);
+
+    StudyEnvironment studyEnvironment =
+        studyEnvironmentService
+            .findByStudy(studyShortcode, envName)
+            .orElseThrow(() -> new NotFoundException("Study environment not found"));
```

Comment:
since we've already validated the study exists in authUserToStudy, this should be an IllegalStateException